File: Algorithms/Implementation/Halloween Sale.php
<?php

/*
 * Complete the 'howManyGames' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER p
 *  2. INTEGER d
 *  3. INTEGER m
 *  4. INTEGER s
 */

function howManyGames($p, $d, $m, $s) {
    # Return the number of games you can buy
    $games = 0;
    while ($s >= 0) {
        if ($p > $s) {
            break;
        }
        $s -= $p;
        $games++;
        $p = max($p - $d, $m);
    }
    return $games;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$first_multiple_input = explode(' ', rtrim(fgets(STDIN)));

$p = intval($first_multiple_input[0]);
$d = intval($first_multiple_input[1]);
$m = intval($first_multiple_input[2]);
$s = intval($first_multiple_input[3]);
$answer = howManyGames($p, $d, $m, $s);

fwrite($fptr, $answer . "\n");
fclose($fptr);

File: Algorithms/Implementation/Halloween Sale.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'howManyGames' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER p
     *  2. INTEGER d
     *  3. INTEGER m
     *  4. INTEGER s
     */

    public static int howManyGames(int p, int d, int m, int s)
    {
        // Return the number of games you can buy
        int count = 0;
        while (s >= p)
        {
            s -= p;
            p -= d;
            count++;
            if (p < m)
            {
                p = m;
            }
        }
        return count;
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string[] firstMultipleInput = Console.ReadLine().TrimEnd().Split(' ');

        int p = Convert.ToInt32(firstMultipleInput[0]);
        int d = Convert.ToInt32(firstMultipleInput[1]);
        int m = Convert.ToInt32(firstMultipleInput[2]);
        int s = Convert.ToInt32(firstMultipleInput[3]);

        int answer = Result.howManyGames(p, d, m, s);

        textWriter.WriteLine(answer);
        textWriter.Flush();
        textWriter.Close();
    }
}
